### PR TITLE
Fixes ActionList css

### LINKED ISSUE #8459: ActionList suffix has unexpected margin on the left when being wrapped
## Issue summary

I'm working on a component that uses up a lot of horizontal space and occasionally wraps the actionList suffix.
image

We would like it if left margin wasn't visible and if the badge could remain aligned with the content, currently we don't have control over this

## Expected behavior

suffix and content are left aligned when wrapped 
image

## Actual behavior
There is a margin on the left of the suffix
image

## Steps to reproduce the problem

1. Create an action list component where one of the items has both `content` and `suffix` filled example
```
        <ActionList
          items={[
            {
              content: "Charge American Express •••• 0005",
              suffix: <Badge>Selected at checkout</Badge>
            },
            {
              content: "Charge Mastercard •••• 4444"
            }
          ]}
        />
```
2. Add this component into a context that has limited width like `<div style={{ width: "300px"><ActionList ...></div>`

## Reduced test case

https://codesandbox.io/s/boring-moon-vbmcy5?file=/App.js

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 10.28.2
- Browser: Chrome
- Device: Macbook M1
- Operating System: OSX

### PR DESCRIPTION
**Pull request by Ipriyankrajai**
### WHY are these changes introduced?

Fixes #8459  

### WHAT is this pull request doing?
Making minor changes in css in ActionList. Using gap property instead of Padding.

Four components are wrapped in flex box with flex-wrap: wrap.
using gap property to solve the issue.   

## Before
Screenshot from 2023-03-02 13-07-53
As we can se when content is overflow. There is a paddingleft to suffix.

## After
Screenshot from 2023-03-02 13-12-49
Using gap property wont add any padding to Suffix.

 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by Ipriyankrajai** [comment]
Hey @alex-page, I've signed CLA. Can anyone review this from Shopify team?

File: .changeset/flat-masks-sniff.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': patch
+---
+
+Fix issue 8459. Made changes for actionlist item use css gap property instead on padding right.
```
Comment (chloerice) [inline .changeset/flat-masks-sniff.md]:
```suggestion
Fixed `ActionList` item `suffix` having extra padding when wrapped
```

**Review by chloerice — APPROVED** [review]
Thanks for contributing this bug fix @lpriyankrajai! 🙌🏽

File: polaris-react/src/components/ActionList/components/Item/Item.tsx [hunk 2]
```diff
@@ -81,15 +81,15 @@ export function Item({
   );
 
   const suffixMarkup = suffix && (
-    <Box paddingInlineStart="4">
+    <Box>
```
Comment (kyledurand) [inline polaris-react/src/components/ActionList/components/Item/Item.tsx:84]: Can `Box` be removed completely now?

Comment (Ipriyankrajai) [reply polaris-react/src/components/ActionList/components/Item/Item.tsx:84]: Yes @kyledurand, we can remove `Box` as Suffix is already wrapped in `span`.

Comment (Ipriyankrajai) [reply polaris-react/src/components/ActionList/components/Item/Item.tsx:84]: Hey @kyledurand, I made changes as you suggested. Added a commit and pushed it to the main branch(i.e. same branch I created pr). But for some reason, I'm unable to see my commit here. Can you help me with this issue?

Comment (kyledurand) [reply polaris-react/src/components/ActionList/components/Item/Item.tsx:84]: Can you create a new pull request for this change? That will run all the necessary checks including chromatic and make sure we don't cause any visual regressions

**Pull request merged**